src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java

```diff
@@ -482,7 +482,9 @@ protected void _addImplicitConstructorCreators(DeserializationContext ctxt,
         final AnnotationIntrospector intr = ccState.annotationIntrospector();
         final VisibilityChecker<?> vchecker = ccState.vchecker;
         List<AnnotatedWithParams> implicitCtors = null;
-        final boolean preferPropsBased = config.getConstructorDetector().singleArgCreatorDefaultsToProperties();
+        final boolean preferPropsBased = config.getConstructorDetector().singleArgCreatorDefaultsToProperties()
```

Comment:
Is this based on semantics needing changing, or just that it happens to solve the failure case?
My understanding is that for Records, default of `Mode.PROPERTIES` is something we would want, not `Mode.DELEGATING`?

Reply:
>  My understanding is that for Records, default of Mode.PROPERTIES is something we would want, not Mode.DELEGATING?

For when the constructor has single-arg, canonical constructor will already be recognised as properties-based creator by:

https://github.com/FasterXML/jackson-databind/blob/a1e6208fd8c6d1873ae41106eb58f8c3751a12f3/src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java#L1025-L1033

...because only a canonical constructor will have an implicit constructor + `propDef` with implicit name.

This `ConstructorDetector` config would only affect non-canonical constructors (which currently are/should only be considered for delegating creators).

**Additional notes**: Currently the only way to "implicitly" make canonical constructor a delegating creator is via `@JsonValue`:
https://github.com/FasterXML/jackson-databind/blob/a1e6208fd8c6d1873ae41106eb58f8c3751a12f3/src/test-jdk14/java/com/fasterxml/jackson/databind/records/RecordImplicitCreatorsTest.java#L34-L44